doc/tor.md

```diff
@@ -211,3 +211,105 @@ for normal IPv4/IPv6 communication, use:
   Otherwise it is trivial to link them, which may reduce privacy. Onion
   services created automatically (as in section 2) always have only one port
   open.
+
+- In some regions of the world the Tor network is monitored or blocked.
+  If you are in an environment that does not permit direct Tor connections or the use
+  of Tor bridges, then considering the trade-offs, it may not be safe to use Tor.
+
+- For maximum privacy, it is preferable to disable accepting incoming connections.
+
+- Users can download packages with [torsocks](https://gitlab.torproject.org/legacy/trac/-/wikis/doc/torsocks) for better security and privacy.
+
+- Trade-offs involved in using `onlynet=onion`:
+
+  The `onlynet=onion` configuration option can potentially ensure the node attempts to only
+  connect over Tor. It is more private when you combine it with no reachable IPv4/IPv6 address,
+  in particular if you want to broadcast transactions without them being correlatable with your IP.
+
+  **Sybil Attacks**
+
+  On the other hand, if you only make random Tor connections, you're much more vulnerable to Sybil attacks.
+  As Tor addresses may be created at no cost, an attacker can potentially flood the network with many Tor
+  nodes and receive all of the outbound Tor connections an `onlynet=tor` node makes.
+
+  This is significantly less a concern with IPv4/IPv6 (especially with asmap) due to the cost of obtaining IPs in many
+  networks. It's also alleviated if you make `-addnode` connections to trusted peers (even if they're onion addresses).
+
+  If all of your connections are controlled by a Sybil attacker, they can easily prevent you from seeing confirmed transactions
+  and, with more difficulty, even trick your node into falsely reporting a transaction as confirmed on the blockchain with most
+  cumulative “chainwork”.
+
+  **Network Partitioning**
+
+  If too many nodes use `onlynet=onion`, it could become difficult for onion nodes to communicate with clearnet nodes,
+  preventing the Tor network from seeing recent transactions and blocks. It is essential that some nodes access both
+  clearnet and Tor, or use `onlynet=i2p` with `onlynet=onion`.
+
+## Setup Guide: Debian/Ubuntu
```

Comment:
Is this onward not just a duplication of lines 59-149? (Any changes should be edited in there).

Reply:
Lines 59-149 have lot of things in detail mentioned under `<h2>` header "Automatically create a Bitcoin Core onion service". The example here is just the basic steps required to quickly setup Bitcoin Core Onion Service without going in to details and skipping whole documentation or fix things in case confused after reading everything written above.

_What?_

The goal here was to mention 3 examples: Ubuntu, Windows and Ubuntu similar to https://bitcoin.stackexchange.com/questions/98913/how-to-run-bitcoin-core-as-onion-service-on-windows-ubuntu-and-android/

_Why?_

1. There can be few users who refer to this doc but find everything mentioned confusing and give up somewhere in between or miss some important steps.
2. We should avoid users searching for proper documentation in different websites, forums, blogs etc. and everyone should be able to follow the things easily using doc in this repository. Privacy should not be hard to achieve or at least the first steps should be easy and encouraging enough to motivate the user in learning more later.

So, the example is basically to skip all the things mentioned above. Run the commands and setup Bitcoin Core Onion Service in few minutes. Once the user can see that onion address and things working it will motivate him to read all the things mentioned in doc if interested to research more.


_How?_

I have added only Ubuntu in this doc because I was expecting if I try to add everything in one PR it will never get merged. I think I was wrong and PR is anyways not merged and its been months. So, I can add more things or look for more ACKs in this to get merged and do follow up PRs later.

I can remove Line 293 and 294 if required to make the example even shorter and just the basic things one needs to try Bitcoin Core onion service quickly.

Reply:
> We should avoid users searching for proper documentation in different websites, forums, blogs etc. and everyone should be able to follow the things easily using doc in this repository

I agree strongly with this.

However, I still don't understand why the steps are duplicated here (it is written differently, but the steps are the same).
1. edit `torrc` file
2. Set:

> ControlPort 9051
> CookieAuthentication 1
> CookieAuthFileGroupReadable 1

3. Set up cookie authentication (add user to Tor group)

Your section "6. Confirmation" is duplicated in lines 8-16 (## How to see information about your Tor configuration via Bitcoin Core)

Reply:
Will plan on adding examples in a separate PR later. Appreciate your review and feedback.